packages/common/src/hooks/useListViewData/usePagination.ts

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from 'react';
+
+interface PaginationState {
+  page: number;
+  offset: number;
+  first: number;
+  onChangePage: (newPage: number) => void;
+  onChangeFirst: (newFirst: number) => void;
+}
+
+export const usePagination = (initialFirst = 20): PaginationState => {
+  const [first, onChangeFirst] = useState(initialFirst);
+  const [offset, onChangeOffset] = useState(0);
+  const [page, onChangePage] = useState(0);
+
+  useEffect(() => {
+    onChangeOffset(page * first);
+  }, [first, page]);
+
+  return { page, onChangePage, onChangeFirst, offset, first };
+};
```

Comment:
Played with using `useReducer` but seemed overkill with all the boilerplate you need.